client/src/app/components/task-manager/TaskManagerDrawer.tsx

```diff
@@ -0,0 +1,256 @@
+
+import "./TaskManagerDrawer.css";
+
+/** A version of `Task` specific for the task manager drawer components */
+interface TaskManagerTask {
+  id: number;
+
+  createUser: string;
+  updateUser: string;
+  createTime: string;
+  started?: string;
+  terminated?: string;
+
+  name: string;
+  kind: string;
+  addon: string;
+  extensions: string[];
+  state: TaskState;
+  priority: number;
+  applicationId: number;
+  applicationName: string;
+  preemptEnabled: boolean;
+}
+
+const PAGE_SIZE = 20;
+
+interface TaskManagerDrawerProps {
+  ref?: React.ForwardedRef<HTMLElement>;
+}
+
+export const TaskManagerDrawer: React.FC<TaskManagerDrawerProps> = forwardRef(
+  (_props, ref) => {
+    const { isExpanded, setIsExpanded, queuedCount } = useTaskManagerContext();
+    const { tasks } = useTaskManagerData();
+
+    const [expandedItems, setExpandedItems] = useState<number[]>([]);
+
+    const closeDrawer = () => {
+      setIsExpanded(!isExpanded);
+      setExpandedItems([]);
+    };
+
+    return (
+      <NotificationDrawer ref={ref}>
+        <NotificationDrawerHeader
+          title="Task Manager"
+          customText={`${queuedCount} queued`}
+          onClose={closeDrawer}
+        >
+          <Link to="/tasks">View All Tasks</Link>
+        </NotificationDrawerHeader>
+        <NotificationDrawerBody>
+          {tasks.length == 0 ? (
+            <EmptyState variant={EmptyStateVariant.full}>
+              <EmptyStateHeader
+                headingLevel="h2"
+                titleText="There are no queued tasks"
+                icon={<EmptyStateIcon icon={CubesIcon} />}
+              />
+              <EmptyStateBody>
+                No tasks are currently ready, postponed, blocked, pending or
+                running. Completed and cancelled tasks may be viewed on the full
+                task list.
+              </EmptyStateBody>
+              <EmptyStateFooter>
+                <EmptyStateActions>
+                  <Link to="/tasks">View All Tasks</Link>
+                </EmptyStateActions>
+              </EmptyStateFooter>
+            </EmptyState>
+          ) : (
+            <NotificationDrawerList>
+              {tasks.map((task) => (
+                <TaskItem
+                  key={task.id}
+                  task={task}
+                  expanded={expandedItems.includes(task.id)}
+                  onExpandToggle={(expand) => {
+                    setExpandedItems(
+                      expand
+                        ? [...expandedItems, task.id]
+                        : expandedItems.filter((i) => i !== task.id)
+                    );
+                  }}
+                />
+              ))}
+            </NotificationDrawerList>
+          )}
+        </NotificationDrawerBody>
+      </NotificationDrawer>
+    );
+  }
+);
+TaskManagerDrawer.displayName = "TaskManagerDrawer";
+
+const TaskStateToIcon: React.FC<{ task: TaskManagerTask }> = ({ task }) =>
```

Comment:
We need to sync on this - in https://github.com/konveyor/tackle2-ui/pull/1957 I've created a mapper function [taskStateToIcon](https://github.com/rszwajko/tackle2-ui/blob/3ed1dd44038319e3e4742b8d0234b32a83cdd603/client/src/app/components/Icons/taskStateToIcon.tsx#L10) based on icons used for [Analysis task state](https://github.com/rszwajko/tackle2-ui/blob/3ed1dd44038319e3e4742b8d0234b32a83cdd603/client/src/app/components/Icons/IconedStatus.tsx#L60).
Key differences:
1.  `Check` icon - here it's Ready, I've chosen Succeeded - Analyze status uses the same icon for both which is confusing for me. Since the Succeeded state is an end state I would prefer to use the icon there.
2.   `Pause` - here it's Postponed state, no icons in my version - IMHO the meaning is too different. I'd rather skip the icon.
3.  `InProgress` - here it's Running, I've used it for Pending - your version fits better
4.  `Pending` -  it fits nicely (no icon in my version)
5. `Unknown` - here it's fallback, in my version it's only used for "placeholder" states used when no state value is received from the backend (`No Task`, `not_supported`).

Reply:
I opened https://github.com/konveyor/tackle2-ui/issues/1969 to capture this work once all the main task functions are merged.

Reply:
Created  PR: https://github.com/konveyor/tackle2-ui/pull/1984